snap_ci/pinned.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Pin commands to install vagrant, ansible, serverspec
+# Pinning commands will ensure that they are run in each snap-ci stage.
+# This is useful for two reasons:
+#   * rerunning a single stage from a pipeline of many
+#   * keeping the CI suite DRY
+
+
+# Cache and install Vagrant
+vagrant_rpm="${vagrant_rpm:-vagrant_1.7.2_x86_64.rpm}"
+[[ -f ${SNAP_CACHE_DIR}/$vagrant_rpm ]] || wget https://dl.bintray.com/mitchellh/vagrant/$vagrant_rpm -O ${SNAP_CACHE_DIR}/$vagrant_rpm
+[[ -x /usr/bin/vagrant ]] || sudo -E rpm -ivh ${SNAP_CACHE_DIR}/$vagrant_rpm
+# TODO: Check for vagrant plugins before installing them.
+
+# An older version of digital_ocean plugin is used because of an issue with the
+# current version that doesn't support using snapshots.
+# https://github.com/smdahlen/vagrant-digitalocean/issues/187
+#/usr/bin/vagrant plugin install vagrant-digitalocean --plugin-version '0.7.0'
+/usr/bin/vagrant plugin install vagrant-digitalocean
```

Comment:
We aren't using DigitalOcean snapshots for automated tests, so might as well use the fully patched version of the plugin. Down the road it would be great to leverage snapshots for testing version upgrades. Will look into that after we have full coverage—that's why the current stages are named `*_clean.sh`, as opposed to `*_upgrade.sh`.